spec/editor-linter-spec.coffee

```diff
@@ -0,0 +1,41 @@
+describe 'editor-linter', ->
+  EditorLinter = require('../lib/editor-linter')
+  beforeEach ->
+    waitsForPromise ->
+      atom.workspace.destroyActivePaneItem()
+      atom.workspace.open('test.txt')
+  describe '::constructor', ->
+    it "cries when provided argument isn't a TextEditor", ->
+      expect ->
+        new EditorLinter
+      .toThrow("Given editor isn't really an editor")
+      expect ->
+        new EditorLinter(null)
+      .toThrow("Given editor isn't really an editor")
+      expect ->
+        new EditorLinter(55)
+      .toThrow("Given editor isn't really an editor")
+  describe '::onShouldLint', ->
+    it 'ignores instant save requests', ->
+      textEditor = atom.workspace.getActiveTextEditor()
+      editorLinter = new EditorLinter(textEditor)
+      timesTriggered = 0
+      editorLinter.onShouldLint ->
+        timesTriggered++
+      textEditor.save()
+      textEditor.save()
+      textEditor.save()
+      textEditor.save()
+      textEditor.save()
+      expect(timesTriggered).toBe(5)
+      editorLinter.deactivate()
```

Comment:
editorLinter and textEditor are created for both tests. use beforeEach and afterEach to separate setup and teardown from the tests.